Source/ARTRealtimeChannel.h

```diff
@@ -16,50 +16,192 @@ NS_ASSUME_NONNULL_BEGIN
+
+/**
+ * BEGIN CANONICAL PROCESSED DOCSTRING
+ * Deregisters the given listener for the specified event name. This removes an earlier event-specific subscription.
+ *
+ * @param name The event name.
+ * @param listener An event listener object to unsubscribe.
+ * END CANONICAL PROCESSED DOCSTRING
+ */
 - (void)unsubscribe:(NSString *)name listener:(ARTEventListener *_Nullable)listener;
 
+/**
+ * BEGIN CANONICAL PROCESSED DOCSTRING
+ * Retrieves an `ARTPaginatedResult` object, containing an array of historical `ARTMessage` objects for the channel. If the channel is configured to persist messages, then messages can be retrieved from history for up to 72 hours in the past. If not, messages can only be retrieved from history for up to two minutes in the past.
+ *
+ * @param query An `ARTRealtimeHistoryQuery` object.
+ * @param callback A callback for retriving an `ARTPaginatedResult` object with an array of `ARTMessage` objects.
+ * @param errorPtr A reference to the `NSError` object where an error information will be saved in case of failure.
+ *
+ * @return In case of failure returns false and the error information can be retrived via the `error` parameter.
+ * END CANONICAL PROCESSED DOCSTRING
+ */
 - (BOOL)history:(ARTRealtimeHistoryQuery *_Nullable)query callback:(ARTPaginatedMessagesCallback)callback error:(NSError *_Nullable *_Nullable)errorPtr;
 
-- (void)setOptions:(ARTRealtimeChannelOptions *_Nullable)options callback:(nullable ARTCallback)cb;
+/**
+ * BEGIN CANONICAL PROCESSED DOCSTRING
+ * Sets the `-[ARTRealtimeChannelProtocol options]` for the channel. An optional callback may be provided to notify of the success or failure of the operation.
+ *
+ * @param options An `ARTRealtimeChannelOptions` object.
+ * @param callback A success or failure callback function.
+ * END CANONICAL PROCESSED DOCSTRING
+ */
+- (void)setOptions:(ARTRealtimeChannelOptions *_Nullable)options callback:(nullable ARTCallback)callback;
 
+/**
+ * BEGIN CANONICAL PROCESSED DOCSTRING
+ * `ARTRealtimeChannel` implements `ARTEventEmitter` and emits `ARTChannelEvent` events, where a `ARTChannelEvent` is either a `ARTRealtimeChannelState` or an `ARTChannelEventUpdate`.
+ * END CANONICAL PROCESSED DOCSTRING
+ */
 ART_EMBED_INTERFACE_EVENT_EMITTER(ARTChannelEvent, ARTChannelStateChange *)
 
 @end
 
 /**
- ARTRealtimeChannel provides a straightforward API for publishing and subscribing to messages on a channel.
- In order to publish, subscribe to, or be present on a channel, you must first obtain a channel instance via ``ARTRealtime/channels/get``.
+ * BEGIN CANONICAL PROCESSED DOCSTRING
+ * Enables messages to be published and subscribed to. Also enables historic messages to be retrieved and provides access to the `ARTRealtimePresence` object of a channel.
+ * Also implements `ARTEventEmitter` interface and emits `ARTChannelEvent` events, where a `ARTChannelEvent` is either a `ARTRealtimeChannelState` or an `ARTChannelEventUpdate`.
```

Comment:
>It would be good to first define that it's an interface to publish/subscribe to Ably's messaging

Isn't it what the previous line is about "Enables messages to be published and subscribed to...."

Reply:
I guess what I meant is putting something on what 'it is'  before what it 'enables'. For example "ARTRealtimeChannel is the main medium open which Ably messages are published and subscribed to" 
Obviously both mean the same things, but I think because this is such an important part of the whole system, I would still prefer something that starts with 'is'. Please feel free to object and I will resolve the conversation in that case.

Reply:
Anyways it's a canonical string which we don't address here, only their compatibility with this particular repo. I've added this suggestion to the canonical repo issue - https://github.com/ably/canonical-api-reference-prototyping/issues/25